app/services/foreman_discovery/fact_parser.rb

```diff
@@ -0,0 +1,16 @@
+module ForemanDiscovery
+  class FactParser < ::PuppetFactParser
+    def suggested_primary_interface(host)
+      return suggested_primary_interface_without_bootif(host) if host.type != "Host::Discovered"
```

Comment:
Good catch, this should call the ordinary fact parser instead. But I wonder if it's possible that the host is not a discovered host. In my testing, I've never encountered this, therefore I could simply remove this line.